File: frontend/src/auth/roles.ts
import type { Role } from '../types/auth'

// หน้าแรก (dashboard) ของแต่ละบทบาท — ล็อกอินแล้วพาไปที่นี่ และผิดบทบาทจะถูกส่งกลับมาที่นี่
const HOME_PATH: Record<Role, string> = {
  admin: '/admin',
  teacher: '/teacher',
  student: '/student',
}

export function homePathForRole(role: Role): string {
  return HOME_PATH[role]
}

File: backend/accounts/roles.py
from django.db import models


class Role(models.TextChoices):
    """บทบาทระดับระบบของผู้ใช้ (1 บัญชี = 1 บทบาท)

    สิทธิ์ในแต่ละรายวิชาแยกอิสระจากบทบาทนี้ — ดูจาก CourseTeacher / Enrollment ต่อวิชา
    (เช่น ผู้เรียนเป็นผู้ช่วยสอนในวิชาหนึ่งได้)
    """

    ADMIN = "admin", "ผู้ดูแลระบบ"
    TEACHER = "teacher", "ผู้สอน"
    STUDENT = "student", "ผู้เรียน"
